### ISSUE TITLE
Feature Request: Add support for custom OpenAI URL to utilize third-party OpenAI API services

### ISSUE THREAD

**Issue by BUJIDAOVS** [kercre123/wire-pod#323]
I have observed that when integrating with GPT, it defaults to accessing api.openai.com. However, I would like to make API requests from a different URL or proxy url. I kindly request the addition of this functionality.

**Comment by dm94** [comment]
This can be very useful as many local systems have created an API similar to OpenAI so that local models such as llama3 could be used.

**Comment by NonaSuomy** [comment]
I attempted to add this a while ago the issue wire told me is that most models are too wordy for vector to not timeout.

I got it to sort of work with textgen UI by setting the model prompt up with "Your answers have to be 255 characters or less" then forcing the settings into vectors backend to use the basic open API but with the local IP address. It would still go bonkers. Vector would make the WORP sound and fail.

**Comment by green-codes** [comment]
Strongly second this, it would make so much sense for the self-hosted wire-pod to also use a self-hosted LM, or at least make it an option.

**Comment by green-codes** [comment]
> Any news on this? I would really like to use my openai provider with my vector

I believe this is already possible by using the custom provider in the knowledge graph settings. Don't know why the issue isn't closed yet, I might be wrong?
